### PR TITLE
Add reconnect and shutdown to x11 tests on s390x

### PR DESCRIPTION
**Pull request by Soulofdestiny**

### CODE REVIEW

**Comment by Soulofdestiny** [comment]
@coolo Can you have a look?
I've got a working test here: http://10.160.5.109/tests/1533
related backend changes: https://github.com/os-autoinst/os-autoinst/pull/477

File: tests/x11/reboot_gnome.pm [hunk 1]
```diff
@@ -26,6 +26,13 @@ sub run() {
         assert_and_click 'reboot-auth-typed', 'right';    # Extra assert_and_click (with right click) to check the correct number of characters is typed and open up the 'show text' option
         assert_and_click 'reboot-auth-showtext';          # Click the 'Show Text' Option to enable the display of the typed text
         assert_screen 'reboot-auth-correct-password';     # Check the password is correct
+
+        # we need to kill ssh for iucvconn here,
+        # because after pressing return, the system is down
+        if (check_var('BACKEND', 's390x')) {
+            console('iucvconn')->kill_ssh;
```
Comment (coolo) [inline tests/x11/reboot_gnome.pm]: I suggest to map these lines into utility function so you can extend later for other backends. Something along "prepare_for_system_shutdown"

**Comment by Soulofdestiny** [comment]
updated

File: tests/x11/shutdown.pm [hunk 2]
```diff
@@ -51,6 +51,11 @@ sub run() {
         if (get_var("SHUTDOWN_NEEDS_AUTH")) {
             assert_screen 'shutdown-auth', 15;
             type_password;
+
+            # we need to kill all open ssh connections before the system shuts down
+            if (check_var('BACKEND', 's390x')) {
+                console('iucvconn')->kill_ssh;
```
Comment (coolo) [inline tests/x11/shutdown.pm]: use that function here too, that was the purpose of having that function

**Comment by Soulofdestiny** [comment]
updated

File: tests/x11/shutdown.pm [hunk 3]
```diff
@@ -78,7 +81,20 @@ sub run() {
...
+    if (check_var('BACKEND', 's390x')) {
+        # make sure SUT shut down correctly
+        console('x3270')->expect_3270(
+            output_delim => qr/.*SIGP stop.*/,
+            timeout      => 30
+        );
+
+        # deactivate consoles
+        do_stop_vm;
+        assert_shutdown;
```
Comment (coolo) [inline tests/x11/shutdown.pm]: as in both if and else assert_shutdown is the last function, move it completely out of the if

**Comment by coolo** [comment]
I assumed we wouldn't want to deploy this week - so if you want to deploy, go ahead. But the risk is yours

**Comment by Soulofdestiny** [comment]
actually i think there wouldn't be any impact because the backend changes are only related to the assert_shutdown - but anyway we wanted to refactor that backend change as discussed with @coolo  - so i will just open a new PR for the tests and update the backend PR when alpha 3 is done

**Pull request merged**